wagtail/users/templates/wagtailusers/groups/includes/formatted_permissions.html

```diff
@@ -38,30 +40,17 @@ <h2>{% trans "Object permissions" %}</h2>
                         {{ content_perms_dict.delete.checkbox.tag }}
                     {% endif %}
                 </td>
+                <td>
+                    {% if content_perms_dict.other %}
+                    <select name="permissions" multiple>
+                        {% for other_perm in content_perms_dict.other %}
+                        <option value="{{ other_perm.perm.id }}" {% if other_perm.selected %}selected{% endif %}>{% trans other_perm.name %}</option>
+                        {% endfor %}
+                    </select>
+                    {% endif %}
+                </td>
```

Comment:
I have not tested this but it might be better to use a multi-checkbox as below (as per discussion, I agree that using select2 is overkill)

```
                <td>
                    {% if content_perms_dict.other %}
                    <fieldset class="other-permissions">
                        <legend>{% trans "Other permissions" %}</legend> 
                        {% for other_perm in content_perms_dict.other %}     
                        <label class="other-permissions-item">
                            <input type="checkbox" name="permissions" value="{{ other_perm.perm.id }}" {% if other_perm.selected %}checked{% endif %}>
                            {{ other_perm.name }}
                        </label>
                        {% endfor %}
                    </fieldset>
                    {% endif %}
                </td>
```

Also, I do not think the `other_perm.name` needs translation, I would assume that would already be provided as a translated value.

The `legend` can be hidden but should be there for screen readers.

As noted above, the `td` would also need to be not rendered if not needed.